doodle.py

```diff
@@ -65,13 +65,51 @@ def main():
 
 # Parse date string and return date and whether it was relative or not
 def derelativize_date(datestr: str, base=datetime.now()):
-    if datestr == '':
-        return base.replace(hour=0, minute=0, second=0, microsecond=0), True
+    relative = False
 
     if not datestr.startswith('+'):
-        return datetime.strptime(datestr, '%Y-%m-%d')
+        return datetime.strptime(datestr, '%Y-%m-%d'), relative
 
-    return base.replace(hour=0, minute=0, second=0, microsecond=0) + timedelta(days=int(datestr[1:])), True
+    else:
+        relative = True
+        date = base.replace(hour=0, minute=0, second=0, microsecond=0)
+        try:
+            # Try to add date
+            date = date + timedelta(days=int(datestr[1:]))
+        except ValueError:
+            # Leave the date as it is.
+            date = date
+
+        return date, relative
+
+
+def append_dates(
+    days: int, weekday: int, relative: bool,
+    start_date: datetime.date, is_weekdays: bool = False,
+    has_args: bool = True
```

Comment:
I don't like using both `is_weekdays` and `has_args`, as they are not at the same "level". `has_args` is `is_weekdays || is_weekends`, so I'd rather pass those two rather than one primitive and one derivated.